tests/sparseml/transformers/finetune/test_finetune.py

```diff
@@ -53,6 +53,38 @@ def test_oneshot_and_finetune(tmp_path: Path):
     )
 
 
+def test_oneshot_and_finetune_with_tokenizer(tmp_path: Path):
+    recipe_str = "tests/sparseml/transformers/finetune/test_alternate_recipe.yaml"
+    model = "Xenova/llama2.c-stories15M"
+    device = "cuda:0"
+    if not torch.cuda.is_available():
+        device = "cpu"
+    dataset_name = "wikitext"
+    dataset_config_name = "wikitext-2-raw-v1"
+    concatenate_data = True
+    run_stages = True
+    output_dir = tmp_path
+    max_steps = 50
+    splits = {"train": "train[:50%]", "calibration": "train[50%:60%]"}
+    tokenizer = SparseAutoTokenizer.from_pretrained(
+        "Xenova/llama2.c-stories15M",
+    )
+
+    compress(
+        model=model,
+        dataset=dataset_name,
+        dataset_config_name=dataset_config_name,
+        run_stages=run_stages,
+        output_dir=output_dir,
+        recipe=recipe_str,
+        max_steps=max_steps,
+        concatenate_data=concatenate_data,
+        splits=splits,
+        oneshot_device=device,
+        tokenizer=tokenizer,
+    )
```

Comment:
Instead of just testing the tokenizer, can we test loading the model and dataset before calling compress? This unit test should basically be a test of the cmd from the documentation: https://neuralmagic.github.io/docs-v2/get-started/finetune/#training